tests/mail/tests.py

```diff
@@ -359,14 +358,17 @@ def test_to_header(self):
             "Subject",
             "Content",
             "bounce@example.com",
-            ["list-subscriber@example.com", "list-subscriber2@example.com"],
-            headers={"To": "mailing-list@example.com"},
```

Comment:
Please add a test to assert that the new exception raises :)

Reply:
I do not think it is wise to alter an existing test and change it's behavior. The test imo shows why we probably cannot raise an exception but have to prefer `extra_headers` over the init kwargs?

Reply:
Thank you all for reviews. @apollo13 Do you think that is better to handle this as suggested by @David-Wobrock above, e.g. in the same way of from?

Reply:
I don't think so, to me the existing `test_to_header` explicitly tests that `extra_data` overrides the `to` argument (whereas the existing from handling simply ignores the value in `extra_headers` if I read it correctly). So we most likely have to keep this behavior.